crates/example-types/src/block_types.rs

```diff
@@ -245,6 +245,29 @@ pub struct TestBlockHeader {
     pub timestamp: u64,
 }
 
+impl TestBlockHeader {
+    fn new<TYPES: NodeType<BlockHeader = Self>>(
```

Comment:
Can we just put the implementation inside `new_legacy` as in the current code? I think in this way it's clearer whether the caller is using the legacy or marketplace version, and makes it less possible to misuse them.

Reply:
As discussed on Zulip this is actually good.